SWLOR.Game.Server/Service/CombatPoint.cs

```diff
@@ -109,37 +109,56 @@ static void DistributeSkillXP()
                         .Where(x => cpList.ContainsKey(x.Key))
                         .ToDictionary(x => x.Key, y => y.Value);
 
-                    var highestRank = skillsWithCP
-                        .Where(x => x.Key != SkillType.Armor)
-                        .OrderByDescending(o => o.Value.Rank)
-                        .Select(s => s.Value.Rank)
-                        .First();
-
-                    // Base amount of XP is determined by the player's highest-leveled skill rank versus the creature's level.
-                    var delta = npcLevel - highestRank;
-                    var baseXP = Skill.GetDeltaXP(delta);
-                    var totalPoints = (float)cpList.Sum(s => s.Value);
                     var areaBonus = GetLocalInt(GetArea(npc), "AREA_XP_BONUS_PERCENTAGE") * 0.01f;
 
-                    // Each skill used during combat receives a percentage of the baseXP amount depending on the number of combat points earned.
-                    foreach (var (skillType, cp) in cpList)
+                    foreach (CombatPointCategory cpCategory in Enum.GetValues(typeof(CombatPointCategory)))
```

Comment:
I'm a LITTLE concerned about performance here. We probably need to run the profiler on it because maybe it's not an issue. But I think with some smarter caching at boot time we may be able to eliminate the 3+ iterations over the enums. I'll do some profiling on this prior to merging this in and report back.

Reply:
Doesn't look like this is that big of a hit. I'm fine to leave it for now. XP gains look like they're working! Nice job!